### PR TITLE
[Doc Improvement][Update auth-related articles for latest TeamsJS v.2]

### PR DESCRIPTION
**Pull request by erikadoyle**
This is one of several PRs that updates existing Teams conceptual articles for TeamsJS v.2, including:

- Prefacing each edited article with note (*[!INCLUDE [sdk-include](~/includes/sdk-include.md)]*) about TeamsJS changes in v.2 and help links
- Updating API names (inline to text content) with v.2 equivalents
- Updating API reference links (inline to text content) with v.2 equivalents
- Implementing tabs for existing code snippets to show both v.1 and v.2 examples

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/how-to/add-authentication.md](https://github.com/MicrosoftDocs/msteams-docs/blob/29bc24a72d12033e883f5bfd13667773ac0ac951/msteams-platform%2Fmessaging-extensions%2Fhow-to%2Fadd-authentication.md) | 88 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/31be52cd-9a96-4f4c-9c1f-328f8cf78930) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-flow-tab.md](https://github.com/MicrosoftDocs/msteams-docs/blob/29bc24a72d12033e883f5bfd13667773ac0ac951/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-flow-tab.md) | 90 | 8 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/57348a89-e829-46ee-81a2-6749eec5f0bc) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/29bc24a72d12033e883f5bfd13667773ac0ac951/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 89 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/b03e809c-be2a-4a33-8841-f84262f5c6b2) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-silent-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/29bc24a72d12033e883f5bfd13667773ac0ac951/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-silent-aad.md) | 92 | 5 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/fadee67a-2ede-4e75-a608-218a0d127f00) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-tab-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/29bc24a72d12033e883f5bfd13667773ac0ac951/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-tab-aad.md) | 89 | 18 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/2c65b491-697e-4ebf-8b7e-eb9471f283d2) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/how-to/add-authentication.md](https://github.com/MicrosoftDocs/msteams-docs/blob/9883a67cae6c2ff81d71a0d65f10fb48ffb9ae92/msteams-platform%2Fmessaging-extensions%2Fhow-to%2Fadd-authentication.md) | 88 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/569b4d7f-cffd-4acb-b093-3eccc9b11eb6) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-flow-tab.md](https://github.com/MicrosoftDocs/msteams-docs/blob/9883a67cae6c2ff81d71a0d65f10fb48ffb9ae92/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-flow-tab.md) | 90 | 8 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/944c90d5-48c6-44ae-b566-5932fc806148) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/9883a67cae6c2ff81d71a0d65f10fb48ffb9ae92/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 89 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/2e90e1d5-1543-4c55-821e-3a2edb1f77fb) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-silent-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/9883a67cae6c2ff81d71a0d65f10fb48ffb9ae92/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-silent-aad.md) | 92 | 5 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/c32ee3fc-67e4-4dc2-992f-f5c3a41d3f81) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-tab-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/9883a67cae6c2ff81d71a0d65f10fb48ffb9ae92/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-tab-aad.md) | 89 | 18 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/8f11202b-b79d-4f5d-881f-e912bb129337) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by erikadoyle** [comment]
@surbhigupta12 when you have a chance, would you mind reviewing? These updates are all based on the updated  [sample code](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-auth/nodejs/src/views/tab/simple). Thanks!

File: msteams-platform/tabs/how-to/authentication/auth-flow-tab.md [hunk 1]
```diff
@@ -21,13 +23,13 @@ For example, the authentication flow for tabs and bots using Node and the [OAuth
 This section uses Azure AD as an example of a third party OAuth provider for enabling authentication in a tab app.
 
 > [!NOTE]
-> Before showing a **Login** button to the user and calling the `microsoftTeams.authentication.authenticate` API in response to selecting the button, you must wait for the SDK initialization to complete. You can pass a callback to the `microsoftTeams.initialize` API that is called when initialization completes.
+> Before showing a **Login** button to the user and calling the `authentication.authenticate` API in response to selecting the button, you must wait for the SDK initialization to complete. You can chain a `.then()` handler or `await` for the `app.initialize()` function to complete.
 
 ![Tab authentication sequence diagram](~/assets/images/authentication/tab_auth_sequence_diagram.png)
```
Comment (v-abirade) [inline msteams-platform/tabs/how-to/authentication/auth-flow-tab.md:28]: The image is not clear or legible, please provide better image.

**Review by v-abirade — CHANGES_REQUESTED** [review]
Request you to update the changes.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/how-to/add-authentication.md](https://github.com/MicrosoftDocs/msteams-docs/blob/c9344264963ae8de1cfc67e8954a1bbe175e4fe7/msteams-platform%2Fmessaging-extensions%2Fhow-to%2Fadd-authentication.md) | 88 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/691081d2-340e-42cb-b06c-03a1d0e74801) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-flow-tab.md](https://github.com/MicrosoftDocs/msteams-docs/blob/c9344264963ae8de1cfc67e8954a1bbe175e4fe7/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-flow-tab.md) | 90 | 8 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/839da4a8-80f6-44cb-b04c-580ab74e941e) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/c9344264963ae8de1cfc67e8954a1bbe175e4fe7/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 89 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/56725aa8-077d-4ce8-be75-49002d662b96) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-silent-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/c9344264963ae8de1cfc67e8954a1bbe175e4fe7/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-silent-aad.md) | 92 | 5 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/0f2e7b69-a5ae-46ee-84d1-51f720b89bb7) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-tab-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/c9344264963ae8de1cfc67e8954a1bbe175e4fe7/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-tab-aad.md) | 89 | 18 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/ebb91943-5236-4b5b-bd52-60ef8be22157) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/how-to/add-authentication.md](https://github.com/MicrosoftDocs/msteams-docs/blob/048d2007e5e8a8f668a769c9ed64cd5bbfeb2753/msteams-platform%2Fmessaging-extensions%2Fhow-to%2Fadd-authentication.md) | 88 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/e63ce0e3-16ad-405a-943b-940e71b29e6b) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-flow-tab.md](https://github.com/MicrosoftDocs/msteams-docs/blob/048d2007e5e8a8f668a769c9ed64cd5bbfeb2753/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-flow-tab.md) | 90 | 8 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/5e57b567-fb94-4738-b556-f3588706e5fe) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-oauth-provider.md](https://github.com/MicrosoftDocs/msteams-docs/blob/048d2007e5e8a8f668a769c9ed64cd5bbfeb2753/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-oauth-provider.md) | 89 | 9 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/68ea99b2-d23b-43e1-92b9-bbe499428610) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-silent-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/048d2007e5e8a8f668a769c9ed64cd5bbfeb2753/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-silent-aad.md) | 92 | 5 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/3562fcb7-2846-4767-9e64-2689d05b4954) | [:white_check_mark:](# "Successfully processed") |
| [msteams-platform/tabs/how-to/authentication/auth-tab-aad.md](https://github.com/MicrosoftDocs/msteams-docs/blob/048d2007e5e8a8f668a769c9ed64cd5bbfeb2753/msteams-platform%2Ftabs%2Fhow-to%2Fauthentication%2Fauth-tab-aad.md) | 89 | 18 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/99467559-ea9f-429d-b1fa-e8d7cc3ea6b5) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/tabs/how-to/authentication/auth-flow-tab.md (hunk 1 above)
Comment (erikadoyle) [reply msteams-platform/tabs/how-to/authentication/auth-flow-tab.md:28]: Done--text size is larger now

File: msteams-platform/tabs/how-to/authentication/auth-tab-aad.md [hunk 2]
```diff
@@ -58,18 +84,49 @@ microsoftTeams.authentication.authenticate({
...
+* Failing to use `authenticate()` will cause a problem with the pop-up not closing at the end of the sign-in process.
 
 ## Navigate to the authorization page from your pop-up page
 
-When your pop-up page (`/tab-auth/simple-start`) is displayed the following code is run. The main goal of this page is to redirect to your identity provider so the user can sign-in. This redirection could be done on the server side using HTTP 302, but in this case it's done on the client side using with a call to `window.location.assign()`. This also allows `microsoftTeams.getContext()` to be used to retrieve hinting information, which can be passed to Azure AD.
+When your pop-up page (`/tab-auth/simple-start`) is displayed the following code is run. The main goal of this page is to redirect to your identity provider so the user can sign-in. This redirection could be done on the server side using HTTP 302, but in this case it's done on the client side using with a call to `window.location.assign()`. This also allows `app.getContext()` to be used to retrieve hinting information, which can be passed to Azure AD.
```
[the remark this answers is not in the document]
Comment (erikadoyle) [reply msteams-platform/tabs/how-to/authentication/auth-tab-aad.md]: Updated with the suggested text/formatting

**Comment by erikadoyle** [comment]
Thanks @v-abirade, I implemented your suggestions. Would you mind taking another look?

**Review by Wajeed-msft — APPROVED** [review]

**Review by v-abirade — APPROVED** [review]
Approved

**Pull request merged**